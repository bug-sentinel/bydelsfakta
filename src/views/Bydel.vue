<template>
  <div class="main-container">
    <div class="main-container__cards">
      <card-image
        v-for="item in items"
        :key="item.value"
        :id="item.value"
        :kategori="item.options.kategori"
        :tema="item.options.tema"
        :bg-image="item.options.bgImage"
        :bg-color="item.options.bgColor"
        :txt-color="item.options.txtColor"
        :link="`${$route.path}/${item.value}`"
      />
    </div>
    <div class="main-container__map">
      <h3>Bydelskart</h3>
      <leaflet-map :district="geoDistricts" />
    </div>
  </div>
</template>

<script>
import { mapGetters } from 'vuex';
import CardImage from '../components/CardImage.vue';
import LeafletMap from '../components/LeafletVue.vue';
import subpages from '../config/subpages';
import districts from '../config/geoData/districts';

export default {
  name: 'Bydel',
  components: {
    CardImage,
    LeafletMap,
  },

  props: {
    bydel: {
      type: String,
      required: true,
    },
  },

  computed: {
    ...mapGetters(['geoDistricts']),
  },

  data() {
    return {
      items: subpages,
      chosenDistrict: {},
      districts: districts,
    };
  },
};
</script>
