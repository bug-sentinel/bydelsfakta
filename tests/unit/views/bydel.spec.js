import { mount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';

import Bydel from '../../../src/views/Bydel.vue';

describe('Bydel', () => {
  let wrapper = null;

  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(VueRouter);
    const router = new VueRouter();
    wrapper = mount(Bydel, {
      propsData: {
        bydel: 'bydel',
      },
      localVue,
      router,
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('renders bydel-component and finds main-container class', () => {
    expect(wrapper.classes('main-container')).toBe(true);
  });

  test('renders component correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
