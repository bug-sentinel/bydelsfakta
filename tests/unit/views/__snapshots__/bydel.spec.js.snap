// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bydel renders component correctly 1`] = `
<div
  class="main-container"
>
  <div
    class="main-container__cards"
  >
    <div
      class="main-container__item"
    >
      <a
        class="oslo__card-image"
        href="#//folkemengde"
        id="folkemengde"
        style="background-image: url(/img/folkemengde.png); background-color: rgb(255, 129, 116);"
      >
        <span
          class="oslo__category"
          style="background-color: rgb(245, 173, 165);"
        >
          
      Befolkning
    
        </span>
         
        <span
          class="oslo__tema"
        >
          Folkemengde
        </span>
         
        <span
          class="oslo__see-more-text"
        >
          Se mer
        </span>
      </a>
    </div>
    <div
      class="main-container__item"
    >
      <a
        class="oslo__card-image"
        href="#//alder"
        id="alder"
        style="background-image: url(/img/alder.png); background-color: rgb(255, 129, 116);"
      >
        <span
          class="oslo__category"
          style="background-color: rgb(245, 173, 165);"
        >
          
      Befolkning
    
        </span>
         
        <span
          class="oslo__tema"
        >
          Alder
        </span>
         
        <span
          class="oslo__see-more-text"
        >
          Se mer
        </span>
      </a>
    </div>
    <div
      class="main-container__item"
    >
      <a
        class="oslo__card-image"
        href="#//innvandrerbefolkningen"
        id="innvandrerbefolkningen"
        style="background-image: url(/img/innvandring.png); background-color: rgb(255, 129, 116);"
      >
        <span
          class="oslo__category"
          style="background-color: rgb(245, 173, 165);"
        >
          
      Befolkning
    
        </span>
         
        <span
          class="oslo__tema"
        >
          Innvandrer­befolkningen
        </span>
         
        <span
          class="oslo__see-more-text"
        >
          Se mer
        </span>
      </a>
    </div>
    <div
      class="main-container__item"
    >
      <a
        class="oslo__card-image"
        href="#//husholdninger"
        id="husholdninger"
        style="background-image: url(/img/husholdninger.png); background-color: rgb(66, 248, 182);"
      >
        <span
          class="oslo__category"
          style="background-color: rgb(199, 247, 201);"
        >
          
      Bomiljø
    
        </span>
         
        <span
          class="oslo__tema"
        >
          Husholdninger
        </span>
         
        <span
          class="oslo__see-more-text"
        >
          Se mer
        </span>
      </a>
    </div>
    <div
      class="main-container__item"
    >
      <a
        class="oslo__card-image"
        href="#//trangboddhet"
        id="trangboddhet"
        style="background-image: url(/img/trangboddhet.png); background-color: rgb(66, 248, 182);"
      >
        <span
          class="oslo__category"
          style="background-color: rgb(199, 247, 201);"
        >
          
      Bomiljø
    
        </span>
         
        <span
          class="oslo__tema"
        >
          Trangboddhet
        </span>
         
        <span
          class="oslo__see-more-text"
        >
          Se mer
        </span>
      </a>
    </div>
    <div
      class="main-container__item"
    >
      <a
        class="oslo__card-image"
        href="#//eierform"
        id="eierform"
        style="background-image: url(/img/eierform.png); background-color: rgb(66, 248, 182);"
      >
        <span
          class="oslo__category"
          style="background-color: rgb(199, 247, 201);"
        >
          
      Bomiljø
    
        </span>
         
        <span
          class="oslo__tema"
        >
          Eierform
        </span>
         
        <span
          class="oslo__see-more-text"
        >
          Se mer
        </span>
      </a>
    </div>
    <div
      class="main-container__item"
    >
      <a
        class="oslo__card-image"
        href="#//kommunaleboliger"
        id="kommunaleboliger"
        style="background-image: url(/img/kommunale-boliger.png); background-color: rgb(66, 248, 182);"
      >
        <span
          class="oslo__category"
          style="background-color: rgb(199, 247, 201);"
        >
          
      Bomiljø
    
        </span>
         
        <span
          class="oslo__tema"
        >
          Kommunale boliger
        </span>
         
        <span
          class="oslo__see-more-text"
        >
          Se mer
        </span>
      </a>
    </div>
    <div
      class="main-container__item"
    >
      <a
        class="oslo__card-image"
        href="#//flytting"
        id="flytting"
        style="background-image: url(/img/flytting.png); background-color: rgb(66, 248, 182);"
      >
        <span
          class="oslo__category"
          style="background-color: rgb(199, 247, 201);"
        >
          
      Bomiljø
    
        </span>
         
        <span
          class="oslo__tema"
        >
          Flytting
        </span>
         
        <span
          class="oslo__see-more-text"
        >
          Se mer
        </span>
      </a>
    </div>
    <div
      class="main-container__item"
    >
      <a
        class="oslo__card-image"
        href="#//bygningstyper"
        id="bygningstyper"
        style="background-image: url(/img/bygningstyper.png); background-color: rgb(66, 248, 182);"
      >
        <span
          class="oslo__category"
          style="background-color: rgb(199, 247, 201);"
        >
          
      Bomiljø
    
        </span>
         
        <span
          class="oslo__tema"
        >
          Bygningstyper
        </span>
         
        <span
          class="oslo__see-more-text"
        >
          Se mer
        </span>
      </a>
    </div>
    <div
      class="main-container__item"
    >
      <a
        class="oslo__card-image"
        href="#//boligpriser"
        id="boligpriser"
        style="background-image: url(/img/boligpriser.png); background-color: rgb(66, 248, 182);"
      >
        <span
          class="oslo__category"
          style="background-color: rgb(199, 247, 201);"
        >
          
      Bomiljø
    
        </span>
         
        <span
          class="oslo__tema"
        >
          Boligpriser
        </span>
         
        <span
          class="oslo__see-more-text"
        >
          Se mer
        </span>
      </a>
    </div>
    <div
      class="main-container__item"
    >
      <a
        class="oslo__card-image"
        href="#//levekaar"
        id="levekaar"
        style="background-image: url(/img/levekaar.png); background-color: rgb(248, 198, 107);"
      >
        <span
          class="oslo__category"
          style="background-color: rgb(244, 206, 138);"
        >
          
      Levekår
    
        </span>
         
        <span
          class="oslo__tema"
        >
          Levekår
        </span>
         
        <span
          class="oslo__see-more-text"
        >
          Se mer
        </span>
      </a>
    </div>
  </div>
   
  <div
    class="main-container__map"
  >
    <h3>
      Bydelskart
    </h3>
  </div>
</div>
`;
